test/unit/caret_test.js

```diff
@@ -0,0 +1,129 @@
+import sinon from 'sinon';
+import {expect} from 'chai';
+import installCaret from '../../src/caret';
+
+describe('Caret', () => {
+  describe('with no document', () => {
+    it('throws an error', () => {
+      expect(installCaret.bind(this, null)).to.throw(/Caret does not have access to document/);
```

Comment:
It is, but I was having issue with the error throwing. This is the only way it would actually catch. Seems stupid to me... but IDK.